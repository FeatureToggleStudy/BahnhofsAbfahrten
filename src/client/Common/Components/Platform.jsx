// @flow
import { cancelled, delayed } from 'style/mixins';
import cc from 'classnames';
import React from 'react';
import withStyles, { type StyledProps } from 'react-jss';

type OwnProps = {|
  cancelled?: boolean,
  scheduled?: string,
  real: string,
|};
type Props = StyledProps<OwnProps, typeof styles>;

const Platform = ({ classes, cancelled, scheduled, real }) => (
  <span
    className={cc({
      [classes.cancelled]: cancelled,
      [classes.delayed]: scheduled && scheduled !== real,
    })}
  >
    {real}
  </span>
);

const styles = {
  cancelled,
  delayed,
};

export default withStyles<Props>(styles)(Platform);
