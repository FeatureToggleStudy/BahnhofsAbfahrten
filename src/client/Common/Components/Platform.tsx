import cc from 'classnames';
import React from 'react';
import useStyles from './Platform.style';

type OwnProps = {
  className?: string;
  cancelled?: boolean;
  scheduled?: string;
  real?: string;
};
type Props = OwnProps;

const Platform = ({ className, cancelled, scheduled, real }: Props) => {
  const classes = useStyles();

  return (
    <span
      className={cc(className, {
        [classes.cancelled]: cancelled,
        [classes.changed]: scheduled && scheduled !== real,
      })}
    >
      {real}
    </span>
  );
};

export default Platform;
