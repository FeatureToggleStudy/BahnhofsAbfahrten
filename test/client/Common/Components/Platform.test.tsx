import { render } from 'testHelper';
import Platform from 'Common/Components/Platform';

describe('Platform', () => {
  it('No platform provided', () => {
    const { container } = render(Platform);

    expect(container).toBeEmpty();
    expect(container).toMatchSnapshot();
  });
  it('scheduled === real', () => {
    const { container } = render(Platform, {
      real: '1',
      scheduled: '1',
    });

    expect(container.innerHTML).toBe('1');
    expect(container).toMatchSnapshot();
  });
  it('scheduled === real & cancelled', () => {
    const { container } = render(Platform, {
      real: '1',
      scheduled: '1',
      cancelled: true,
    });

    expect(container.innerHTML).toBe('1');
    expect(container).toHaveStyle('text-decoration: line-through');
    expect(container).toMatchSnapshot();
  });
  it('scheduled !== real & cancelled', () => {
    const { container, theme } = render(Platform, {
      real: '1',
      scheduled: '2',
      cancelled: true,
    });

    expect(container.innerHTML).toBe('1');
    expect(container).toHaveStyle(`
    text-decoration: line-through;
    color: ${theme.colors.red};`);
    expect(container).toMatchSnapshot();
  });
  it('scheduled !== real', () => {
    const { container, theme } = render(Platform, {
      real: '1',
      scheduled: '2',
    });

    expect(container.innerHTML).toBe('1');
    expect(container).not.toHaveStyle(`text-decoration: line-through;`);
    expect(container).toHaveStyle(`color: ${theme.colors.red};`);
    expect(container).toMatchSnapshot();
  });
});
