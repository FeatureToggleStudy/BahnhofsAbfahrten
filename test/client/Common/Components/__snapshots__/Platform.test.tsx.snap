// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platform No platform provided 1`] = `
<span
  class=""
/>
`;

exports[`Platform scheduled !== real & cancelled 1`] = `
<span
  class="Platform-cancelled-0 Platform-changed-0"
>
  1
</span>
`;

exports[`Platform scheduled !== real 1`] = `
<span
  class="Platform-changed-0"
>
  1
</span>
`;

exports[`Platform scheduled === real & cancelled 1`] = `
<span
  class="Platform-cancelled-0"
>
  1
</span>
`;

exports[`Platform scheduled === real 1`] = `
<span
  class=""
>
  1
</span>
`;
